Dart/src/com/jetbrains/lang/dart/ide/generation/CreateConstructorWithNamedArgumentsFix.java

```diff
@@ -0,0 +1,77 @@
+// Copyright 2000-2021 JetBrains s.r.o. and contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package com.jetbrains.lang.dart.ide.generation;
+
+import com.intellij.codeInsight.template.Template;
+import com.intellij.codeInsight.template.TemplateManager;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.NlsContexts;
+import com.jetbrains.lang.dart.DartBundle;
+import com.jetbrains.lang.dart.psi.DartClass;
+import com.jetbrains.lang.dart.psi.DartComponent;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Iterator;
+import java.util.Set;
+
+public class CreateConstructorWithNamedArgumentsFix extends BaseCreateMethodsFix<DartComponent> {
+  public CreateConstructorWithNamedArgumentsFix(@NotNull final DartClass dartClass) {
+    super(dartClass);
+  }
+
+  @Override
+  protected void processElements(@NotNull final Project project,
+                                 @NotNull final Editor editor,
+                                 @NotNull final Set<DartComponent> elementsToProcess) {
+    final TemplateManager templateManager = TemplateManager.getInstance(project);
+    anchor = doAddMethodsForOne(editor, templateManager, buildFunctionsText(templateManager, elementsToProcess), anchor);
+  }
+
+  @Override
+  protected @NotNull @NlsContexts.Command String getCommandName() {
+    return DartBundle.message("dart.generate.constructor.with.named.arguments");
+  }
+
+  @Override
+  @NotNull
+  protected String getNothingFoundMessage() {
+    return ""; // can't be called actually because processElements() is overridden
+  }
+
+  protected Template buildFunctionsText(TemplateManager templateManager, Set<DartComponent> elementsToProcess) {
+    final Template template = templateManager.createTemplate(getClass().getName(), DART_TEMPLATE_GROUP);
+    template.setToReformat(true);
+
+    //noinspection ConstantConditions
+    template.addTextSegment(myDartClass.getName());
+    template.addTextSegment("({");
+    for (Iterator<DartComponent> iterator = elementsToProcess.iterator(); iterator.hasNext(); ) {
+      DartComponent component = iterator.next();
+
+      if (component.isFinal() && !isNullable(component)) {
+        template.addTextSegment("required ");
+      }
+
+      template.addTextSegment("this.");
+      //noinspection ConstantConditions
+      template.addTextSegment(component.getName());
+      if (iterator.hasNext()) {
+        template.addTextSegment(",");
+      }
+    }
+    template.addTextSegment(",});");
+    template.addEndVariable();
+    template.addTextSegment(" "); // trailing space is removed when auto-reformatting, but it helps to enter line break if needed
+    return template;
+  }
+
+  private static Boolean isNullable(DartComponent component) {
```

Comment:
Did not find a way to check if a field is nullable, thus implemented this very crude "isNullable" method. Any feedback on this is welcome.